src/parse/asp/builtins.go

```diff
@@ -1022,17 +1024,23 @@ func subrepo(s *scope, args []pyObject) pyObject {
 	} else if args[PathArgIdx] != None {
 		root = string(args[PathArgIdx].(pyString))
 	}
+
+	// Base name
+	subrepoName := path.Join(s.pkg.Name, name)
+
+	// State
 	var state *core.BuildState
 	if args[ConfigArgIdx] != None {
-		state = s.state.ForSubrepo(name, path.Join(s.pkg.Name, string(args[ConfigArgIdx].(pyString))))
+		state = s.state.ForSubrepo(subrepoName, path.Join(s.pkg.Name, string(args[ConfigArgIdx].(pyString))))
```

Comment:
`s.CurrentSubrepo` should include the package as well.

Reply:
Nice! I spotted this but wasn't 100% sure how it should work. I think this makes sense though.